### PR TITLE
WIP: summary tab tables (data only). Note: all rules are not clear so…

### PR DESCRIPTION
**Pull request by evans-g-crsj**
… I left identity functions as placeholder for coming work.

### CODE REVIEW

File: src/pages/variantEntity/Summary.css [hunk 1]
```diff
@@ -0,0 +1,18 @@
+.grid-top-container {
```
Comment (evans-g-crsj) [inline src/pages/variantEntity/Summary.css:1]: Temporary : it was already there before the pr. I just renamed the file

File: src/pages/variantEntity/TabSummary.tsx [hunk 2]
```diff
@@ -1,49 +1,216 @@
...
+const orderGenes = (mSymbolToConsequences: SymbolToConsequences) => {
+  if (!mSymbolToConsequences || Object.keys(mSymbolToConsequences).length === 0) {
+    return [];
+  }
+  return Object.entries(mSymbolToConsequences).map(([, values]) => {
+    return { ...values };
+  });
+};
+
+const orderConsequences = (consequences: Consequence[]) => [...consequences];
```
Comment (evans-g-crsj) [inline src/pages/variantEntity/TabSummary.tsx:57]: rules are not clear yet from spec. However, consequences will have to be ordered so I left a placeholder

File: src/pages/variantEntity/TabSummary.tsx [hunk 3]
```diff
@@ -1,49 +1,216 @@
...
+const makeTables = (rawConsequences: Consequence[]) => {
+  if (!rawConsequences || rawConsequences.length === 0) {
+    return [];
+  }
+  const symbolToConsequences = groupConsequencesBySymbol(rawConsequences);
+  const orderedGenes = orderGenes(symbolToConsequences);
+  return orderConsequencesForTable(orderedGenes);
+};
+
+const columns = [
```
Comment (evans-g-crsj) [inline src/pages/variantEntity/TabSummary.tsx:82]: all data is there, but needs to be formatted. I made sure that all is visible.

File: src/pages/variantEntity/Summary.tsx [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+  }
+
+  return (
+    <Card>
+      <div className={'grid-top-container'}>
+        <div className={'grid-top-container-item'}>
+          {SummaryItem('Chr', variant.chromosome)}
+          {SummaryItem('Start', variant.start)}
+          {SummaryItem('Allele Alt.', variant.start)}
+          {SummaryItem('Allele Réf.', variant.reference)}
```
Comment (francisl) [inline src/pages/variantEntity/Summary.tsx:37]: why not a react component?

File: src/pages/variantEntity/Summary.tsx [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+
+import { VariantEntity } from 'store/graphql/variants/models';
+import { Card } from 'antd';
+
+type SummaryProps = {
+  variant: VariantEntity | undefined;
+};
+
+const SummaryItem = (field: string, value: string) => {
+  return (
```
Comment (francisl) [inline src/pages/variantEntity/Summary.tsx:17]: no need for a return here

File: src/pages/variantEntity/TabClinical.tsx [hunk 6]
```diff
@@ -14,7 +14,17 @@ type OwnProps = {
...
+  const { loading, data, error } = useTabClinicalData(variantId);
+
+  if (error) {
+    return (
+      <Result
+        status="500"
+        title="Server Error"
+        subTitle="An error has occured and we are not able to load content at this time."
+      />
+    );
```
Comment (francisl) [inline src/pages/variantEntity/TabClinical.tsx:26]: used at 3 places at least it has react the magic number to extract

File: src/pages/variantEntity/TabSummary.tsx [hunk 7]
```diff
@@ -1,49 +1,216 @@
...
+  if (!tableGroups || tableGroups.length === 0) {
+    return [];
+  }
+
+  return tableGroups.map((tableGroup: TableGroup) => {
+    const consequences = tableGroup.consequences;
+    return {
+      ...tableGroup,
+      consequences: orderConsequences(consequences),
+    };
```
Comment (francisl) [inline src/pages/variantEntity/TabSummary.tsx:68]: two types of returns, [] and {}

File: src/pages/variantEntity/TabSummary.tsx [hunk 8]
```diff
@@ -1,49 +1,216 @@
...
+          {makeTables(consequences).map((tableData: TableGroup, index: number) => {
+            const symbol = tableData.symbol;
+            const omim = tableData.omim;
+            const biotype = tableData.biotype;
+            const orderedConsequences = tableData.consequences;
+            return (
+              <Table
+                key={index}
+                title={() => [symbol, omim, biotype].join(' ')}
+                bordered
```
Comment (francisl) [inline src/pages/variantEntity/TabSummary.tsx:199]: no bordered table.

We need to wrap it in a container as in Figma

**Review by francisl — CHANGES_REQUESTED** [review]
I'll not block because of deadline, but please consider comment. 
The tables should not be bordered. That one need to be fixed. The require and extra container. I'll add them

File: src/pages/variantEntity/Summary.tsx (hunk 4 above)
Comment (evans-g-crsj) [reply src/pages/variantEntity/Summary.tsx:37]: took from another branch. not my dev

File: src/pages/variantEntity/TabClinical.tsx (hunk 6 above)
Comment (evans-g-crsj) [reply src/pages/variantEntity/TabClinical.tsx:26]: corrected

File: src/pages/variantEntity/TabSummary.tsx (hunk 8 above)
Comment (evans-g-crsj) [reply src/pages/variantEntity/TabSummary.tsx:199]: no, one return type

File: src/pages/variantEntity/Summary.tsx (hunk 5 above)
Comment (evans-g-crsj) [reply src/pages/variantEntity/Summary.tsx:17]: took from another branch. not my dev

**Review by francisl — APPROVED** [review]

**Pull request merged**